actors/market/tests/market_actor_test.rs

```diff
@@ -1278,6 +1279,295 @@ fn active_deals_multiple_times_with_different_providers() {
+        let f = |rt: &mut MockRuntime, d: &mut DealProposal| {
+            // with these two equal provider collatreal min is 5/100 * deal size
+            let power = StoragePower::from_i128(1 << 50).unwrap();
+            rt.set_circulating_supply(power);
+            let deal_size = 2048; // default deal size used
+            let policy = Policy::default();
+            let provider_min = (deal_size * policy.prov_collateral_percent_supply_num as u64)
+                / (policy.prov_collateral_percent_supply_denom as u64);
+            d.provider_collateral = TokenAmount::from(provider_min - 1);
+        };
+        assert_deal_failure(true, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
+    }
+
+    #[test]
+    fn negative_client_collateral() {
+        let f = |_rt: &mut MockRuntime, d: &mut DealProposal| {
+            d.client_collateral = TokenAmount::from(-1);
+        };
+        assert_deal_failure(true, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
+    }
+
+    #[test]
+    fn client_collateral_greater_than_max_collateral() {
+        let f = |_rt: &mut MockRuntime, d: &mut DealProposal| {
+            d.client_collateral = TOTAL_FILECOIN.clone() + 1;
+        };
+        assert_deal_failure(true, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
+    }
+
+    #[test]
+    fn client_does_not_have_enough_balance_for_collateral() {
+        let owner_addr = Address::new_id(OWNER_ID);
+        let provider_addr = Address::new_id(PROVIDER_ID);
+        let worker_addr = Address::new_id(WORKER_ID);
+        let client_addr = Address::new_id(CLIENT_ID);
+
+        let f = |rt: &mut MockRuntime, d: &mut DealProposal| {
+            add_participant_funds(rt, client_addr, d.client_balance_requirement() - 1);
+            add_provider_funds(
+                rt,
+                d.provider_collateral.clone(),
+                provider_addr,
+                owner_addr,
+                worker_addr,
+            );
+        };
+        assert_deal_failure(false, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
+    }
+
+    #[test]
+    fn provider_does_not_have_enough_balance_for_collateral() {
+        let owner_addr = Address::new_id(OWNER_ID);
+        let provider_addr = Address::new_id(PROVIDER_ID);
+        let worker_addr = Address::new_id(WORKER_ID);
+        let client_addr = Address::new_id(CLIENT_ID);
+
+        let f = |rt: &mut MockRuntime, d: &mut DealProposal| {
+            add_participant_funds(rt, client_addr, d.client_balance_requirement());
+            add_provider_funds(
+                rt,
+                d.provider_collateral.clone() - 1,
+                provider_addr,
+                owner_addr,
+                worker_addr,
+            );
+        };
+        assert_deal_failure(false, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
+    }
+
+    #[test]
+    fn unable_to_resolve_client_address() {
+        let f = |_rt: &mut MockRuntime, d: &mut DealProposal| {
+            d.client = Address::new_id(1);
```

Comment:
Keep this test case `client_address_does_not_exist' but then also add one matching the go test behavior of using a bls address.